app/navigation/AssetNavigator.scala

```diff
@@ -71,17 +72,31 @@ class AssetNavigator @Inject() (config: FrontendAppConfig) extends Navigator {
   private def assetsCompletedRoute(draftId: String): Call =
     Call(GET, config.registrationProgressUrl(draftId))
 
-  private def addAssetsRoute(draftId: String)(answers: UserAnswers): Call =
+  private def addAssetsRoute(draftId: String)(answers: UserAnswers): Call = {
     answers.get(AddAssetsPage) match {
+      case Some(AddAssets.YesNow) if nonTaxableNonEeaBusinessRoute(draftId).isDefinedAt(answers) => nonTaxableNonEeaBusinessRoute(draftId)(answers)
```

Comment:
I think this is simpler?
```suggestion
            case Some(AddAssets.YesNow) if !answers.isTaxable && answers.assets.nonEEABusiness.isDefined => nonTaxableNonEeaBusinessRoute(draftId, answers)
```

Along with this change:
```scala anotatte
  private def nonTaxableNonEeaBusinessRoute(draftId: String, answers: UserAnswers): Call = {
    val numNonEeaAssets = answers.assets.nonEEABusiness.get.size
    val index = numNonEeaAssets - 1
    NameController.onPageLoad(index, draftId)
  }
```